ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/QueryCompactor.java

```diff
@@ -97,6 +97,7 @@ void runCompactionQueries(HiveConf conf, String tmpTableName, StorageDescriptor
     Util.overrideConfProps(conf, compactionInfo, tblProperties);
     String user = compactionInfo.runAs;
     SessionState sessionState = DriverUtils.setUpSessionState(conf, user, true);
+    sessionState.setCompactionSession(true);
```

Comment:
Should this flag be cleared after the compaction queries are finished?

Reply:
I don't think so, it should not cause any issues. Worker runs only compaction queries.

Reply:
SemanticAnalyzer is used for all queries (even if Worker is a compaction-only class), and SessionState is shared between queries, isn't it?
so I'm afraid the fact that Worker is a compaction class doesn't make us sure that compactionSession is handled properly, it should be confirmed that a user cannot hack ranger policies by doing something like:
1. run a compaction manually -> SessionState.setCompactionSession(true)
2. run a query that involves ranger filtering -> what is SessionState.get().isCompactionSession() here?

Reply:
When a manual compaction is added it's just queued, the worker's session is completely independent from the user's one

Reply:
okay, I think as the second occurrence (so the one after the compaction) of the query below:
```
select * from masking_test_n_compact;
```
returns the masked result in the qtest, the proper behavior is kind of confirmed
so it's fine I think, are you fine with this @kasakrisz ?

Reply:
shouldn't we use queryState instead? 
````
Driver driver = creator.createDriver(
          new QueryState.Builder()
              .withHiveConf(conf)
              .withGenerateNewQueryId(true)
              .nonIsolated()
              .build());
````
SessionStateUtil
  public static Optional<QueryState> getQueryState(Configuration conf) {
    return Optional.ofNullable(SessionState.get()).map(ss -> ss.getQueryState(HiveConf.getQueryId(conf)));
  }

Reply:
However the Worker's session is used only to execute compaction related queries, for safety I added the line to restore the flag after executing the compaction queries.